pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -459,7 +459,7 @@ func (configgen *ConfigGeneratorImpl) buildGatewayHTTPRouteConfig(node *model.Pr
 				gatewayRoutes[gatewayName][vskey] = routes
 			}
 			for _, hostname := range intersectingHosts {
-				if vHost, exists := vHostDedupMap[hostname]; exists {
+				if vHost, exists := vHostDedupMap[host.Name(strings.ToLower(hostname.String()))]; exists {
```

Comment:
not sure i understand, why not convert before this line

Reply:
Do not understand. Are you suggesting  ` host:= host.Name(strings.ToLower(hostname.String())`  - some thing like this for readability?

Reply:
And does nameToServiceMap[hostname] matter

Reply:
nameToServiceMap - Does not matter

Reply:
>Do not understand. Are you suggesting host:= host.Name(strings.ToLower(hostname.String()) - some thing like this for readability?

I mean we convert this before, so all the code within the loop will use small cased hostname